autoload/lsp/omni.vim

```diff
@@ -107,8 +109,25 @@ function! s:handle_omnicompletion(server_name, complete_counter, data) abort
     endif
 endfunction
 
-function! lsp#omni#get_kind_text(completion_item) abort
-    return has_key(a:completion_item, 'kind') && has_key(s:kind_text_mappings, a:completion_item['kind']) ? s:kind_text_mappings[a:completion_item['kind']] : ''
+function! lsp#omni#get_kind_text(completion_item, ...) abort
+    let l:server = get(a:, 1, '')
+    if empty(l:server) " server name 
+        let l:completion_item_kinds = s:default_completion_item_kinds
+    else
+        if !has_key(s:completion_item_kinds, l:server)
+            let l:server_info = lsp#get_server_info(l:server)
+            if has_key (l:server_info, 'config') && has_key(l:server_info['config'], 'completion_item_kinds')
+                let s:completion_item_kinds[l:server] = s:default_completion_item_kinds
```

Comment:
@clason 
Just thinking about this (a little late I know): does this make a copy of the dict or just a reference to it?
I think the following line still changes `s:default_completion_item_kinds`?

Reply:
Yes, you're right! Sorry, vimL newbie.

It's probably best to write this as
```viml
let s:completion_item_kinds[l:server] = extend(copy(s:default_completion_item_kinds), l:server_info['config']['completion_item_kinds']))
```

Reply:
That should work. I should've caught this before merging, I guess we're both newbies :smiley: 
Can you send PR?

Reply:
Done #529 Thanks for catching that!